### ISSUE TITLE
Cyclic dependency detected when dealing with Domain Events

### ISSUE THREAD

**Issue by ecostanzi** [moduliths/moduliths#249]
I'm trying moduliths tests after looking at Spring IO talk [Spring for the architecturally curious developer](https://www.youtube.com/watch?v=zrqupmaIsto). I'm working on this repository: https://github.com/ecostanzi/modulith-tests

I'm having an issue with Modulith tests detecting a cyclic dependency due to classpath dependency of event classes.

```
org.moduliths.model.Violations: - Cycle detected: Slice modulea -> 
                Slice moduleb -> 
                Slice modulea
  1. Dependencies of Slice modulea
    - Method <com.example.modulithtest.modules.modulea.ServiceA.handleEvent(com.example.modulithtest.modules.moduleb.EventB)> has parameter of type <com.example.modulithtest.modules.moduleb.EventB> in (ServiceA.java:0)
  2. Dependencies of Slice moduleb
    - Constructor <com.example.modulithtest.modules.moduleb.ServiceB.<init>(com.example.modulithtest.modules.modulea.ServiceA, org.springframework.context.ApplicationEventPublisher)> has parameter of type <com.example.modulithtest.modules.modulea.ServiceA> in (ServiceB.java:0)
    - Field <com.example.modulithtest.modules.moduleb.ServiceB.serviceA> has type <com.example.modulithtest.modules.modulea.ServiceA> in (ServiceB.java:0)
   
```

I usually put the event classes in the package of the module that publishes the events. That said I found myself in this situation.
- `ServiceB` depends on `ServiceA`
- `ServiceB` publishes `EventB` 
- `ServiceA` is subscribed to `EventB`

Modulith test doesn't like this configuration because it detects a cyclic dependency. This is of course true at the classpath level, but the fact that `ServiceA` depends on `EventB` is a way to decouple the two modules and avoid cyclic dependencies.

Looking at the generated canvas, I would say that `ModuleB` uses `ModuleA`, `ModuleA` listens to `ModuleB`, but I wouldn't say that `ModuleA` uses `ModuleB`.
 
![components-Modules](https://user-images.githubusercontent.com/3602612/185365531-2c0693cf-c62f-4890-b401-959ed5a42e9a.png)

Is there a flaw in my design? Is there a way to avoid the test failure without putting all the DomainEvents in a `shared` package?

Thanks for the support, I really like this tool!

**Comment by ecostanzi** [comment]
I just realized the project has been moved to https://github.com/spring-projects-experimental/spring-modulith. Let me know if I have to move the issue too.

**Comment by odrotbohm** [comment]
Why would you both bean-depend B -> A *and* have A listen to events of B? That creates the cycle and Moduliths is telling you about that 🙃. You need to decide, which of the two types of dependencies you want to establish and then stick to that one.

The "uses" dependency type is supposed to describe a Spring bean dependency, so if that shows up, there is indeed a bug somewhere.

**Comment by odrotbohm** [comment]
This one should be fixed in the latest snapshots for 1.3.1. I see your sample project produce the correct component diagram now. Feel free to give the snapshots a try.
